pgsql-flexible-server.tf

```diff
@@ -163,3 +169,29 @@ resource "null_resource" "set-user-permissions-additionaldbs" {
     azurerm_postgresql_flexible_server_database.pg_databases
   ]
 }
+
+resource "azurerm_monitor_diagnostic_setting" "pgsql_logs" {
+  name                       = azurerm_postgresql_flexible_server.pgsql_server.name
+  target_resource_id         = azurerm_postgresql_flexible_server.pgsql_logs.id
+  log_analytics_workspace_id = module.log_analytics_workspace_id. workspace_id
+
+    dynamic "log" {
+    for_each = toset(var.flexible_server_logs)
+    content {
+      category = log.value
+      enabled  = true
+      retention_policy {
+        enabled = true
+        days    = data.azurerm_log_analytics_workspace.log_analytics_workspace.retention_in_days
```

Comment:
this won't work as we aren't using a datasource here, you'll have to hardcode something sensible.